students/TacSPx/lesson03/mailroom.py

```diff
@@ -0,0 +1,156 @@
+            ("K.J. Wright", [21663.23, 300.87, 100432.0]),
+            ("Curt Warner", [663.23, 300.87, 10432.0]),
+            ]
+
+# Processing  --------------------------------------------------------------- #
+# Task 2:  Menu choices
+# Prompt the user to choose from a menu of 3 actions: “Send a Thank You”, “Create a Report” or “quit”.
+###############################################
+#   "Menu of Choices"
+###############################################
+
+def main():
+    '''
+    Menu of Choices:  Thank You, Report, Create a Report
+    :return:
+    '''
+
+    while (True):
+        print("""
+        \n**** XYZ Nonprofit Charity ****\n
+        Menu of Options
+        1) Send a "Thank You"
+        2) Create a Report
+        3) Exit Program
+        """)
+        strChoice = str(input("Which option would you like to perform? [1 to 3] - "))
+        print()  # added a new line for spacing looks
+        if (strChoice.strip() == '1'):
+            thank_you()
+        elif (strChoice.strip() == '2'):
+            report()
+        elif (strChoice.strip() == '3'):
+            exit_program()
+        else:
+            print("\nPlease choose only 1, 2, or 3!")
+
+###############################################
+#   #1 Send a "Thank You"
+###############################################
+def thank_you():
+    '''
+    Thank You:  Will append donations to existing donors, or adds new donors with donations.
+    User can see list of existing donors by typing 'list'.  Once donation is entered script will create
+    custom thank you letter/ email.
+    :return:
+    '''
+    while (True):
+        donor_name = input("Type 'List' to see all names in Database or\nType First and Last Name: ").title()
+        if (donor_name.lower() == 'list'):
+            # define format row
+            row = "{Name:<18s}".format
+            print("\nDONOR REGISTRY:")
+            donor_sort = (sorted(donor_info))
+            for donor in donor_sort:
+                donor_lst = (row(Name=donor[0]))
+                print(donor_lst)
+            input("Press Enter to Continue")
+            print()
+            continue
+
+        if (donor_name.lower() != 'list'):
+            donor_amount = int(input("Enter Amount of Donation: $"))
+            # if name is in list append dollar amount to donors bucket
+            for i in donor_info:
+                if i[0] == donor_name:
+                    i[1].append(donor_amount)
+                    print(f"'{donor_name}' is in registry added new donation amount!")
+                    break
+                # if name not in list add new name to donor info list
+                else:
+                    donor_info.append((donor_name, [donor_amount]))
+                    print(f"New donor name '{donor_name}' added to registry!")
+                    # print(donor_info) - for testing
+                    break
+            input("\nPress Enter for Thank You letter:\n")
+            print(f'''Dear {donor_name},
```

Comment:
This isn't summing new donations to existing donations.  Can you see why?

Reply:
The issue here is the for loop above on line 82. Let's step through it.

Let's say your list has ['bob', 'joe', 'kat'] and we would like to add 'natasha'

in the first loop iteration you have two checks (if and else), 

first one `if i[0] == donor_name:` which will be false because 'natasha' isn't in your list and so it will go into `else` and add "natasha" and `break` out of the loop

the second time you run a program with intent to update donations to natasha, it will run through the same logic again.. however ``if i[0] == donor_name:`` will be FALSE _again_ well because 'bob' != 'natasha' so it will go into `else` block and ADD natasha _again_ and `break` out...

you see the issue there?

so how do we fix it? what you really want is an else statement in a different place:
```
for i in donor_info:
    if i[0] == donor_name:
        # append
        break
else:
    # add
```
in above loop, you will continue checking EACH donor name against the one from the user input, _if_ it matches you will break, if it will not match any of the list entries, it will go into the `else` block of the `FOR` loop but only after exhausting all list entries.